File: mgyoutube-multitech/api-dotnet/modules/DefaultSearchesModuleImpl.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using api_dotnet.apimodel;
using api_dotnet.repos;

namespace api_dotnet.modules
{
    public class DefaultSearchesModuleImpl : SearchesModule
    {
        public SearchesDataRepo SearchesDataRepo { get; set; }
        public UserDataRepo UserDataRepo { get; set; }
        public DefaultSearchesModuleImpl(UserDataRepo userDataRepo, SearchesDataRepo searchesDataRepo)
        {
            UserDataRepo = userDataRepo;
            SearchesDataRepo = searchesDataRepo;
        }

        public async Task AddSearchToParent(string parentUsername, string searchPhrase)
        {
            Console.WriteLine("AddSearchToParent, parentUsername=" + parentUsername + ", searchPhrase=" + searchPhrase);
            User parentUser = await this.UserDataRepo.GetUserByUsername(parentUsername);

            if (parentUser == null)
            {
                Console.WriteLine("AddSearchToParent, parentUsername=" + parentUsername + " parentUser is null so throwing UserNotFoundException");
                throw new UserNotFoundException(parentUsername);
            }

            Console.WriteLine("AddSearchToParent, parentUsername=" + parentUsername + ", found parent=" + parentUser);

            await this.SearchesDataRepo.AddSearchToParentUser(parentUser.userId, searchPhrase);
        }

        public async Task<List<string>> GetSavedSearchesForParent(string parentUsername)
        {
            User parentUser = await this.UserDataRepo.GetUserByUsername(parentUsername);

            if (parentUser == null)
            {
                throw new UserNotFoundException(parentUsername);
            }

            List<string> searches = await this.SearchesDataRepo.GetSearchesForParentUser(parentUser.userId);

            Console.WriteLine("getSavedSearchesForParent for " + parentUsername + " returning searches.size " + searches.Count);

            return searches;
        }
    }
}

File: mgyoutube-multitech/api-java/src/main/java/com/djpedesen/mgyoutube/api_java/modules/DefaultSearchesModuleImpl.java
package com.djpedesen.mgyoutube.api_java.modules;

import java.util.List;

import com.djpedesen.mgyoutube.api_java.apimodel.User;
import com.djpedesen.mgyoutube.api_java.repos.SearchesDataRepo;
import com.djpedesen.mgyoutube.api_java.repos.UserDataRepo;

public class DefaultSearchesModuleImpl implements SearchesModule {

	private UserDataRepo userDataRepo;
	private SearchesDataRepo searchesDataRepo;

	public DefaultSearchesModuleImpl(final UserDataRepo userDataRepo, final SearchesDataRepo searchesDataRepo) {
		this.setUserDataRepo(userDataRepo);
		this.setSearchesDataRepo(searchesDataRepo);
	}

	public void setUserDataRepo(final UserDataRepo userDataRepo) {
		this.userDataRepo = userDataRepo;
	}

	public void setSearchesDataRepo(final SearchesDataRepo searchesDataRepo) {
		this.searchesDataRepo = searchesDataRepo;
	}

	@Override
	public List<String> getSavedSearchesForParent(final String parentUsername) throws Exception {

		final User parentUser = this.userDataRepo.getUserByUsername(parentUsername);

		if (parentUser == null) {
			throw new UserNotFoundException(parentUsername);
		}

		final List<String> searches = this.searchesDataRepo.getSearchesForParentUser(parentUser.userId);

		System.out.println(
				"getSavedSearchesForParent for " + parentUsername + " returning searches.size " + searches.size());

		return searches;
	}

	@Override
	public void addSearchToParent(final String parentUsername, final String searchPhrase) throws Exception {
		final User parentUser = this.userDataRepo.getUserByUsername(parentUsername);

		if (parentUser == null) {
			throw new UserNotFoundException(parentUsername);
		}

		this.searchesDataRepo.addSearchToParentUser(parentUser.userId, searchPhrase);
	}

}
